### ISSUE TITLE
No exception / error logged though starting fails

### ISSUE THREAD

**Issue by webdevilopers** [Phobetor/rabbitmq-supervisor-bundle#36]
We use capistrano for deployment. It seems to have some problems with symlinks or similar.
But we could break it down to the main problem.

When we config our supervisor path to e.g. `tmp` or below the Symfony kernel root dir everything works as expected and supervisor gets started:

```
rabbit_mq_supervisor:
    supervisor_instance_identifier:     "%supervisor_instance_identifier%"
    paths:
        #workspace_directory:            "/tmp/supervisor/%supervisor_instance_identifier%/"
        workspace_directory:            "%kernel.root_dir%/../../../supervisor/%supervisor_instance_identifier%/"
```

But if we keep the `supervisor` dir inside the `app` folder for instance the starting fails.

In both cases the files are created correctly.

When I run the command on the server with the non-working dir I don't get any response - no log entries - no process starting:

```
SYMFONY_ENV=prod app/console rabbitmq-supervisor:control start
```

Though the exception has been added in #22 there still seems to be something missing.

Is there a `debug` or `trace` argument I can use? Or some way to hack into this:
* https://github.com/Phobetor/rabbitmq-supervisor-bundle/blob/master/Services/Supervisor.php#L98-L100

**Comment by webdevilopers** [comment]
Just recognized we were running the following version:

```
"phobetor/rabbitmq-supervisor-bundle": "2.0.0",
```

It thought that release
https://github.com/Phobetor/rabbitmq-supervisor-bundle/releases/tag/2.0.0
would alread include the Logger seen here:
https://github.com/Phobetor/rabbitmq-supervisor-bundle/blob/master/Services/Supervisor.php#L98-L100

But it doesn't. I must use `dev-master` to have it?!

**Comment by webdevilopers** [comment]
Finally after using `dev-master` I get some logs in Symfony:

```
[2018-03-10 10:18:27] app.DEBUG: Executing: supervisorctl --configuration=/var/www/vhosts/.../htdocs/releases/20180310091756/app/../supervisor/sps_intranet/supervisord.conf status [] []
[2018-03-10 10:18:27] app.CRITICAL: supervisorctl returns code: Misuse of shell builtins [] []
[2018-03-10 10:18:27] app.DEBUG: Output: error: <class 'socket.error'>, AF_UNIX path too long: file: /usr/lib64/python2.7/socket.py line: 224  [] []
[2018-03-10 10:18:27] console.ERROR: Exception thrown while running command "rabbitmq-supervisor:rebuild". Message: "Misuse of shell builtins" {"exception":"[object] (Phobetor\\RabbitMqSupervisorBundle\\Exception\\ProcessException(code: 2): Misuse of shell builtins at /var/www/vhosts/.../htdocs/releases/20180310091756/vendor/phobetor/rabbitmq-supervisor-bundle/Phobetor/RabbitMqSupervisorBundle/Services/Supervisor.php:67)","command":"rabbitmq-supervisor:rebuild","message":"Misuse of shell builtins"} []
[2018-03-10 10:18:27] console.ERROR: Command "rabbitmq-supervisor:rebuild" exited with code "2" {"command":"rabbitmq-supervisor:rebuild","code":2} []
```

**Comment by webdevilopers** [comment]
Similar error messages here:
* https://github.com/spatie/laravel-backup/issues/92
* https://github.com/spatie/laravel-backup/issues/285

**Comment by webdevilopers** [comment]
At the bottom line it looks like the error occurs if the configured directories are none-permanent.
While a deployment with Capistrano a new folder is added under "releases". After deployment finished it is symlinked to "current".
Maybe the link does not exist for a short period of time - thought the absolute path inside the error looks correct and should work.

**Comment by webdevilopers** [comment]
My provider just told me that the actual ".." could be causing the message. It does not seem to be allowed when setting the path for supervisor.

I guess that means that I cannot use "dynamic" paths inside the symfony configuration for this bundle.
I will use absolute paths.

**Comment by webdevilopers** [comment]
Now i finally recognized that onky the `.conf` file was created. `log`, `pid` and `.sock` were missing.
That will cause the exception.

What I cannot tell is what causes them not to be created. Well I know that it does not work in newly created and later symlinked folders.

Maybe a permission problem while deployment?

**Comment by Phobetor** [comment]
Hi @webdevilopers

Thanks for all the information and first of all sorry for the confusion with 2.0.0 vs. dev-master. There is a new up-to-date version [3.0.0](https://github.com/Phobetor/rabbitmq-supervisor-bundle/releases/tag/3.0.0) out now.
You can use that version and `console rabbitmq-supervisor:rebuild --wait-for-supervisord` to get error output.

The `.conf` file is created by the commands of this bundle. The `.log`, `.pid` and `.sock` files are created by `supervisord` on start. If the latter do not exist the start of `supervisord` may have failed. It is hard to tell why that happened.

You write "it does work when using ../../../". That sounds like the other path is invalid for some reason. Maybe the directory does not exist or is not writable for the user that runs the command?

**Comment by Phobetor** [comment]
As there is no new information on this issue and it should be fixed in 3.0.0 I will close it. Feel free to open a new one if there are still problems with the current version.

**Comment by webdevilopers** [comment]
I tried to deploy with the new `--wait-for-supervisord` flag but get no new error messages.

> The .conf file is created by the commands of this bundle. The .log, .pid and .sock files are created by supervisord on start.
> If the latter do not exist the start of supervisord may have failed. It is hard to tell why that happened.

I agree. It really looks like a problem with the server process. The bundle looks fine actually.

Thanks for the help so far.

**Comment by webdevilopers** [comment]
I could now break down and re-produce our issue.

When our capistrano deployment executed the rebuild command it gives no error and supervisor is not restarted on the remote server.
When I use your new flag "wait-for-supervisord" supervisor IS restarted. But capistrano freezes waiting for a response infinitely.

However similar happens if I execute the commands outside of the deployment directly on the server.
Without the flag supervisor is not started. With the flag it is start and it DOES NOT HANG UP.

The conflict seems to be running the rebuild while being in the capistrano deployment process.

Ever stumbled upon this with other deplyoment tools?

Maybe these consumer options could help?
* https://github.com/Phobetor/rabbitmq-supervisor-bundle/pull/12#issuecomment-271088914

**Comment by vvasiloi** [comment]
Just had the same issue: seems like it fails when it cannot read or write the configuration files.
I was running  `rabbitmq-supervisor:rebuild --wait-for-supervisord` command with `www-data` user in a Docker container. Executing `chown -R www-data:www-data` on the configuration folder solved the issue.

**Comment by webdevilopers** [comment]
Thanks @vvasiloi for your feedback. I no longer feel alone now! :)
TLDR I think we moved the configuration files to a different location where indeed no permission problems can occur. It worked out so far. Since then we never touched it again.

**Comment by vvasiloi** [comment]
If the issue is caused by not being able to read/write configuration files then it should throw a specific error, not an unclear one like "Misuse of shell builtins".
The bundle should check if it can read from/write to the configuration folder before running the supporvisor commands.
Will try to reproduce the issue and come with a PR later.
